### PR TITLE
Update testing rig

### PR DESCRIPTION
**Pull request by levithomason**
This PR is a breakout of the massive rewrite to drop jQuery, #206.

### CODE REVIEW

File: build/karma.conf.babel.js [hunk 1]
```diff
@@ -10,13 +10,12 @@ module.exports = (karmaConfig) => {
     singleRun: !argv.watch,
     reporters: ['mocha'],
     files: [
-      './node_modules/babel-polyfill/dist/polyfill.js',
-      './node_modules/phantomjs-polyfill/bind-polyfill.js',
```
Comment (levithomason) [inline build/karma.conf.babel.js:14]: New phantom no longer needs bind polyfill (or we stopped using it).

File: build/karma.conf.babel.js [hunk 2]
```diff
@@ -10,13 +10,12 @@ module.exports = (karmaConfig) => {
...
+      require.resolve('babel-polyfill/browser'),
       './test/tests.bundle.js',
     ],
     frameworks: ['mocha'],
     preprocessors: {
-      '**/*.bundle.js': ['webpack', 'sourcemap'],
+      '**/*.bundle.js': ['webpack'],
```
Comment (levithomason) [inline build/karma.conf.babel.js:18]: no longer required with move to `karma-webpack-with-fast-source-maps`.

File: build/webpack.config.js [hunk 3]
```diff
@@ -34,12 +34,15 @@ const webpackHotMiddlewareEntry = 'webpack-hot-middleware/client?' + _.map({
   quiet: false,           // Set to true to disable all console logging.
 }, (val, key) => `&${key}=${val}`).join('')
 
-const APP_ENTRY = paths.docsSrc('DocsApp.js')
+const APP_ENTRY = [
+  require.resolve('babel-polyfill'),
+  paths.docsSrc('DocsApp.js'),
+]
```
Comment (levithomason) [inline build/webpack.config.js:40]: This is the doc site entry, we now add the babel-polyfill here.

File: package.json [hunk 4]
```diff
@@ -80,20 +80,17 @@
     "highlight.js": "8.9.1",
     "html-webpack-plugin": "^2.14.0",
     "imports-loader": "^0.6.4",
-    "json": "^9.0.3",
```
Comment (levithomason) [inline package.json:83]: cruft

**Comment by eanplatter** [comment]
🍂 once the branch is up to date with master and it passes CI

**Comment by ghost** [comment]
⛵

**Pull request merged**